tcp_check/tests/test_tcp_check.py

```diff
@@ -97,18 +101,153 @@ def test_response_time(aggregator):
+        ),
+        pytest.param(
+            'some-hostname',
+            common.DUAL_STACK_GETADDRINFO_IPV4,
+            [addr_tuple('ip1', socket.AF_INET), addr_tuple('ip2', socket.AF_INET6), addr_tuple('ip3', socket.AF_INET6)],
+            True,
+            3,
+            id='Dual IPv4/IPv6: string hostname, IPv4 resolution, multiple_ips True',
+        ),
+        pytest.param(
+            'another-hostname',
+            common.DUAL_STACK_GETADDRINFO_IPV6,
+            [addr_tuple('ip1', socket.AF_INET6), addr_tuple('ip2', socket.AF_INET), addr_tuple('ip3', socket.AF_INET6)],
+            True,
+            3,
+            id='Dual IPv4/IPv6: string hostname, IPv6 resolution, multiple_ips True',
+        ),
+        pytest.param(
+            'localhost',
+            common.SINGLE_STACK_GETADDRINFO_LOCALHOST_IPV4,
+            [addr_tuple('127.0.0.1', socket.AF_INET)],
+            False,
+            1,
+            id='Single stack IPv4: localhost, IPv4 resolution, multiple_ips False',
+        ),
+        pytest.param(
+            'localhost',
+            common.SINGLE_STACK_GETADDRINFO_LOCALHOST_IPV4,
+            [addr_tuple('127.0.0.1', socket.AF_INET), addr_tuple('ip2', socket.AF_INET)],
+            True,
+            2,
+            id='Single stack IPv4: localhost, IPv4 resolution, multiple_ips True',
+        ),
+        pytest.param(
+            'another-hostname',
+            common.SINGLE_STACK_GETADDRINFO_IPV4,
+            [addr_tuple('ip1', socket.AF_INET)],
+            False,
+            1,
+            id='Single stack IPv4: string hostname, IPv4 resolution, multiple_ips False',
+        ),
+        pytest.param(
+            'another-hostname',
+            common.SINGLE_STACK_GETADDRINFO_IPV4,
+            [addr_tuple('ip1', socket.AF_INET), addr_tuple('ip2', socket.AF_INET), addr_tuple('ip3', socket.AF_INET)],
+            True,
+            3,
+            id='Single stack IPv4: string hostname, IPv4 resolution, multiple_ips True',
+        ),
+    ],
+)
+def test_hostname_resolution(
+    aggregator, monkeypatch, hostname, getaddrinfo, expected_addrs, multiple_ips, service_checks_count
+):
+    """
+    Test that string hostnames get resolved to ipv4 address format properly.
+    """
+    instance = deepcopy(common.INSTANCE)
+    instance['host'] = hostname
+    instance['multiple_ips'] = multiple_ips
+    check = TCPCheck(common.CHECK_NAME, {}, [instance])
+
+    monkeypatch.setattr('socket.getaddrinfo', mock.Mock(return_value=getaddrinfo))
+    monkeypatch.setattr(check, 'connect', mock.Mock(side_effect=[None, None, None]))
+
+    expected_tags = [
+        "instance:UpService",
+        "target_host:{}".format(hostname),
+        "port:80",
+        "foo:bar",
+        "address:{}".format(expected_addrs[0].address),
+    ]
+
+    check.check(instance)
+
+    assert check._addrs == expected_addrs
+    # at the end of check.check() run, check.socket_type will be assigned to the last addr's socket type
```

Comment:
The check may iterate over both IPv4 and IPv6 addresses when `multiple_ips: True`. Is there a better way to assert the `check.socket_type` value as it gets reassigned based on each address's socket_type?

Reply:
Why do you want to assert the `check.socket_type` value? You are already asserting on the addresses that are being used. A possible improvement I can think of is to assert that the calls to `connect` look as expected (with methods like [`assert_called_with`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with).

Reply:
Good idea, thanks!